### PR TITLE
do not escape xml with invalid strings

### LINKED ISSUE #9534: japanese model produces XML error
Issue overview
--------------
This Japanese [IDF](https://github.com/NREL/EnergyPlus/files/9105845/in_2210.idf.txt) produces an [XML](https://github.com/NREL/EnergyPlus/files/9105838/in_2210Table.xml.txt) with an error. The other files (e.g. HTML) are unaffected. The middle character in Zone `電気室 19` becomes unreadable, see line 5575 for first instance. As far as I can tell the character `気` is supported by [Unicode CJK](https://www.unicode.org/charts/PDF/U4E00.pdf). This model originally came from Revit and was updated from 9.1.0 to 22.1.0 with a couple changes to get it to run.

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version) = Windows 10
 - Version of EnergyPlus (if using an intermediate build, include SHA) = 22.1.0
 - Unmethours link or helpdesk ticket number = NA

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by marklemay**
Pull request overview
---------------------
 - Fixes #9534
 - xml was previously being escaped with invalid strings like `&deg;`.  This is a minimal fix:
    * In xml output other multibyte Unicode characters can bump into the ACSII characters that do need to be escaped mangling the string. I believe the xml would still be valid according to the xml specification.
    * In html output some Unicode will still be mangled, though I believe the output will still be valid html.

### Pull Request Author
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [x] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/OutputReportTabular.cc [hunk 1]
```diff
@@ -17334,27 +17334,27 @@ std::string ConvertToEscaped(std::string const &inString, bool isXML) // Input S
             s += "&lt;";
         } else if (c == '>') {
             s += "&gt;";
-        } else if (c == char(176)) {
+        } else if (c == char(176) && !isXML) {
             s += "&deg;";
```
Comment (marklemay) [inline src/EnergyPlus/OutputReportTabular.cc]: `&deg;` is not a valid xml escape sequence.

**Review by JasonGlazer — APPROVED** [review]
The code changes look good.

**Comment by JasonGlazer** [comment]
Just waiting for CI to be all green as expected.

**Comment by JasonGlazer** [comment]
All green, ready to merge.

**Comment by Myoldmopar** [comment]
Fine with me, thanks @marklemay and @JasonGlazer, merging!

**Pull request merged**